### ISSUE TITLE
Eclipse Che on Windows 8.1

### ISSUE THREAD

**Issue by bobpaige** [eclipse-che/che#2620]
I am trying to evaluate Che in my development group but can't get it to run reliably.  Some of this is known (was explained to me as a limitation while at JavaOne last week). Some appears to be boot2docker.

Known Issue: Don't Sleep
1. install boot2docker from scratch
2. follow basic che instructions

and it works. But if I put my machine to sleep, Che never comes back.

Instability: Boot2Docker
The gentleman I spoke to at JavaOne suggested the workaround for the "sleep" issue is to install the "che" script and shutdown the che server before sleeping the machine. An extra step, but ok.

But I can't even run the "che" script without Bash installed, and the tiny linux install that boot2docker is based on doesn't include bash. So you install it with a simple command. Until you restart boot2docker. Then any new stuff you've installed goes away and you need to re-install bash, and che. But even then, I can't get che-server to run again.

**Reproduction Steps:**
1. Start VirtualBox
   $ docker-machine start
2. SSH into the docker
   $ docker-machine ssh default
3. Install bash
   $ tce-load -wi bash
4. Install che and make it executable
   $ sudo -i
   $ curl -sL https://raw.githubusercontent.com/eclipse/che/master/che.sh > /usr/local/bin/che
   $ chmod +x /usr/local/bin/che
   $ exit
5. Start che-server
   $ che start

Log output:
INFO: ECLIPSE CHE: Found image codenvy/che-server:latest
INFO: ECLIPSE CHE: Starting container...
INFO: ECLIPSE CHE: Server logs at "docker logs -f che-server"
INFO: ECLIPSE CHE: Server booting...
INFO: ECLIPSE CHE: Booted and reachable
INFO: ECLIPSE CHE: Ver: 4.7.2
INFO: ECLIPSE CHE: Use: http://192.168.99.100:8080
INFO: ECLIPSE CHE: API: http://192.168.99.100:8080/swagger
1. Open che in web browser
   At this point, the browser says the server is not responding.
2. Check che-server logs
   $  docker logs -f che-server

About 1 screen of log output, but the last line is:
2016-09-27 19:57:20,953[main]             [INFO ] [o.a.catalina.startup.Catalina 642]   - Server startup in 7113 ms

*_che info output *_
INFO: ---------------------------------------
INFO: -------------   CLI INFO   ------------
INFO: ---------------------------------------
INFO:
INFO: ---------  PLATFORM INFO  -------------
INFO: CLI DEFAULT PROFILE       = not set
INFO: CHE_VERSION               = latest
INFO: CHE_CLI_VERSION           = latest
INFO: CHE_UTILITY_VERSION       = nightly
INFO: DOCKER_INSTALL_TYPE       = boot2docker
INFO: DOCKER_HOST_IP            = 192.168.99.100
INFO: IS_NATIVE                 = NO
INFO: IS_WINDOWS                = NO
INFO: IS_DOCKER_FOR_WINDOWS     = NO
INFO: IS_DOCKER_FOR_MAC         = NO
INFO: IS_BOOT2DOCKER            = YES
INFO: HAS_DOCKER_FOR_WINDOWS_IP = NO
INFO: IS_MOBY_VM                = NO
INFO: HAS_CHE_ENV_VARIABLES     = NO
INFO: HAS_TEMP_CHE_PROPERTIES   = NO
INFO: IS_INTERACTIVE            = YES
INFO: IS_PSEUDO_TTY             = YES
INFO:
INFO: ---------------------------------------
INFO: ---------   LAUNCHER INFO  ------------
INFO: ---------------------------------------
INFO:
INFO: ---------  PLATFORM INFO  -------------
INFO: DOCKER_INSTALL_TYPE       = boot2docker
INFO: DOCKER_HOST_OS            = Boot2Docker 1.12.1 (TCL 7.2); HEAD : ef7d0b4 - Thu Aug 18 21:18:06 UTC 2016
INFO: DOCKER_HOST_IP            = 192.168.99.100
INFO: DOCKER_HOST_EXTERNAL_IP   = not set
INFO: DOCKER_DAEMON_VERSION     = 1.12.1
INFO:
INFO:
INFO: --------- CHE INSTANCE LIST  ----------
INFO:
INFO:
INFO: ----  CURRENT COMMAND LINE OPTIONS  ---
INFO: CHE_PORT                  = 8080
INFO: CHE_VERSION               = latest
INFO: CHE_RESTART_POLICY        = no
INFO: CHE_USER                  = root
INFO: CHE_HOST_IP               = 192.168.99.100
INFO: CHE_LOG_LEVEL             = info
INFO: CHE_DEBUG_SERVER          = false
INFO: CHE_DEBUG_SERVER_PORT     = 8000
INFO: CHE_HOSTNAME              = 192.168.99.100
INFO: CHE_DATA_FOLDER           = /home/user/che
INFO: CHE_CONF_FOLDER           = not set
INFO: CHE_LOCAL_BINARY          = not set
INFO: CHE_SERVER_CONTAINER_NAME = che-server
INFO: CHE_SERVER_IMAGE_NAME     = codenvy/che-server

If I restart the virtual machine (docker-machine restart) I'm back to square one as far as needing to install bash and che, but the server is still inaccessible.

I only started using Che last week. Maybe this is a Windows 8 issue? The default when installing Docker on windows only supports Windows 10. Maybe the Win8 Docker is unsupported? Maybe I'm missing some key steps during install that preserves data?

My first attempt at using Che several weeks ago was with Yatta. That never worked, though looking back I suspect it was the same issues I'm having here; no bash, no che.

**Comment by TylerJewell** [comment]
Chuckle :). So I can see the issue.  I am on my phone right now and will need to be on a computer to work through it all.   But you need bash installed native on Windows, not inside of the VM.  Installing Docker Toolbox for Windows should have you install Git for Windows.  And that includes bash.  

If you did that install then surprised that bash is not in your path.  When you open a windows batch shell and you type bash, what happens?

UPDATE:
Ok - So the part that is getting you tripped up is that bash and boot2docker are different requirements. When you are on Windows 8 with boot2docker, we use that behind the scenes to power Che & your workspaces. However, the bash scripts that start / stop Che are done outside of that container.

So to resolve your issue, you want to:
1: Verify that you installed Docker Toolbox on your windows box (should be ok since you already have boot2docker)
2: Verify that you can open a windows batch prompt and type "bash" and it is working.
3: Then install the che.sh + che.bat according to the windows [instructions here](https://www.eclipse.org/che/getting-started/download/).  Follow the windows instructions, not the Linu instructions.
4: Then in your windows batch prompt, you can run 'che start' and 'che stop'.  

Before you run 'che start', in the back ground start your docker-machine the way you have done before.  Che will verify that it can use docker.  If your shell has been setup properly, then Che will make use of boot2docker and handle the file mounting along with the ability to start / stop cleanly.

**Comment by ghost** [comment]
@bobpaige you are running Che launcher inside a VM rather than do it from your Windows system. Che actually starts in a VM but for some reason IP auto detection shows the wrong IP. Can you verify that your VM IP is 192.168.99.100?

Otherwise, @TylerJewell 's suggestions will certainly help

**Comment by bobpaige** [comment]
Thank you both; I am indeed an idiot. I assumed that once I had docker/linux running on my machine I would follow the linux instructions, which was false and clearly described on your installation instructions page.

After installing the che.sh and che.bat files in the bash console started by the Docker Quickstart Terminal, it does work great! Bummer I had to install yet another bash shell to get this to work, but hey, it's windows. You live with what works.

**Comment by TylerJewell** [comment]
On my windows system, I have bash on a normal dos prompt - it comes with
docker toolbox, which installed git for windows.  So I don't even need to
open docker quick start terminal for it.

Tyler Jewell | CEO | tyler@​codenvy.​com | 9​78​.8​84​.53​55

**Comment by bobpaige** [comment]
I previously installed cygwin, which worked great and was upgradable.
Then we switched to "TortoiseGit" which installs msysgit and git-bash. Neither of which are compatible with cygwin. So I switched to msysgit.
After I installed docker toolbox I could access the "docker-machine" and "che" commands from my git-bash terminal, but they don't run the same. I can only run them from within the Docker terminal. No biggy; I'll just use that for docker/che commands and use git-bash for all other stuff.

Now that I have this working, I'll open a separate issue about creating complex projects within Che.

Thanks again for the quick and thorough support, and not dealing harshly with someone making stupid mistakes.
